### ISSUE TITLE
🐛 [SAMBA NAS] Unable to upload or rename files in external usb

### ISSUE THREAD

**Issue by mariogarofalo** [dianlight/hassio-addons#171]
### Addon

SambaNAS

### Description

Greetings.

Apologies if this topic has already been covered, but I'm new to HA and I found the Samba NAS to be spectacular for my needs.

However, I have a small problem, and I hope you can help me solve it: I can view the files easily from all devices connected to the network, and I can delete them from the USB memory.
However, if I try to load new files (both from the HA interface and from Windows 10 File Manager) the files seem to be copied, but I can't find them in the destination folder.
Furthermore, if I try to rename one of the files already present on the external memory, the operation is not allowed and I receive an error.
Has anyone had something similar happen and managed to resolve it? Thank you

### Reproduction steps

```bash
1. Go to 'Media' > "My Media" > "usb device"
2. Click on 'upload media'
2. Caricamento non riuscito: Uploaded file is too large (file.mov)
```

### Addon Logs

```bash
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service base-addon-banner: starting
-----------------------------------------------------------
 Add-on: Samba NAS
 Expose Home Assistant disc with SMB/CIFS
-----------------------------------------------------------
 Add-on version: 10.0.2-nas4
 You are running the latest version of this add-on.
 System: Home Assistant OS 10.5  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2023.9.2
 Home Assistant Supervisor: 2023.09.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-automount: starting
[10:58:01] INFO: Protection Mode is false
[10:58:01] INFO: MoreDisk or Automount option found!
---------------------------------------------------
Supported fs: 	ext3	ext2	ext4	squashfs	vfat	msdos	f2fs	fuseblk	ntfs3	exfat	btrfs	xfs
Supported fusefs: 
---------------------------------------------------
[10:58:01] INFO: MediaLibrary option found!
---------------------------------------------------
[10:58:02] INFO: Available Disk Labels:
[10:58:02] INFO: 	MARIOTVSHOW[vfat]
---------------------------------------------------
[10:58:02] INFO: No MoreDisks to mount
[10:58:02] INFO: Automount is Enabled!
---------------------------------------------------
[10:58:02] INFO: Mounting disks:
	MARIOTVSHOW
---------------------------------------------------
[10:58:02] WARNING: Your MARIOTVSHOW is vfat. Permissions and ACL don't works and this is an EXPERIMENTAL support
[10:58:02] INFO: Mount MARIOTVSHOW[vfat] Success!
s6-rc: info: service init-automount successfully started
s6-rc: info: service init-smartd: starting
s6-rc: info: service init-samba: starting
s6-rc: info: service hd-idle: starting
s6-rc: info: service hd-idle successfully started
[10:58:03] INFO: Hostname: HomeAssistantServer
s6-rc: info: service init-smartd successfully started
```

### Addon Config

```bash
workgroup: WORKGROUP
username: 
allow_hosts:
  - 10.0.0.0/8
  - 172.16.0.0/12
  - 192.168.0.0/16
  - 169.254.0.0/16
  - fe80::/10
  - fc00::/7
automount: true
moredisks:
  - "true"
available_disks_log: true
medialibrary:
  enable: true
veto_files:
  - ._*
  - .DS_Store
  - Thumbs.db
  - icon?
  - .Trashes
compatibility_mode: true
wsdd2: true
mqtt_use_legacy_entities: false
autodiscovery: {}
other_users: []
acl: []
interfaces: []
password:
```

### Architecture

amd64

### OS

HaOS

**Comment by dianlight** [comment]
Hi,
 your disk is a VFAT disk so is limited to max 4Gb of file size. If `file.mov` is bigger cant fit. :)

On RPI there are often power problems on USB devices and this can be show as "phantom" writes as you describe.  Check your RPI power status and allarm ( there is a specific sensor on HA ) and if is possible add a power adapter directly on the disk to check. 
Also use USB2 port and not the USB3 can be usefull sometimes.

Also yours problems ( rename for example ) can be vfat and permission incompatibility.  Onestly I never try with vfat disk (  i try to test it as soon as possible to se if there is a bug in the mount code).

If you can I suggest you to change the fs of the disk to a more modern fs like exFat or better to a linux support like ext4.

L.

**Comment by dianlight** [comment]
should be fixed  in the release 12.1.0-nas

**Comment by dianlight** [comment]
Also try to disable timemachine. vfat  don't play well with extended attributes.

```yaml
acl:
  - share: MARIOTVSHOW
     timemachine: false
```

L.
